src/tentaclio/clients/ftp_client.py

```diff
@@ -89,18 +89,42 @@ def scandir(self, **kwargs) -> Iterable[fs.DirEntry]:
         """Scan the connection url to create dir entries."""
         base_url = f"ftp://{self.url.hostname}:{self.port}{self.url.path}/"
         entries = []
-        for mlsd_entry in self.conn.mlsd(self.url.path, facts=["type"]):
-            # https://docs.python.org/3/library/ftplib.html#ftplib.FTP.mlsd
-            file_name = mlsd_entry[0]
-            entry_type = mlsd_entry[1]["type"]
+        try:
+            for mlsd_entry in self.conn.mlsd(self.url.path, facts=["type"]):
+                # https://docs.python.org/3/library/ftplib.html#ftplib.FTP.mlsd
+                file_name = mlsd_entry[0]
+                entry_type = mlsd_entry[1]["type"]
+
+                url = urls.URL(base_url + file_name)
+                if entry_type == "dir":
+                    entries.append(fs.build_folder_entry(url))
+                else:
+                    entries.append(fs.build_file_entry(url))
+        except ftplib.error_perm:
+            # Server doesn't support mlsd
+            lines = self.ftp_dir()
+            file_names = self._parse_ftp_dir(lines)
+            for file_name in file_names:
+                url = urls.URL(base_url + file_name)
+                entries.append((fs.build_file_entry(url)))
+        return entries
 
-            url = urls.URL(base_url + file_name)
-            if entry_type == "dir":
-                entries.append(fs.build_folder_entry(url))
-            else:
-                entries.append(fs.build_file_entry(url))
+    def ftp_dir(self):
```

Comment:
I forgot to mention that circleci will run `pydocstyle` too. It complains about this guy not having documentation. I would probably make it private anyway as not everyone needs to access it.